### DISCUSSION TITLE
AVRdude 7.3 Upload speed decreases from 28.2kB/s to 23.1kB/s

### DISCUSSION THREAD

**Discussion by flyandancexo** [avrdudes/avrdude#1692] [General]
Test done with the same computer and same target MCU, AVRdude 7.3's upload speed drops significantly.  Looks like people updating v7.3 learns how to drive backward. It's exactly the same upload speed of v6.3; I also tested v7.1; v7.1 is good. V7.3 is the version to be skipped for me.

V7.3
![A](https://github.com/avrdudes/avrdude/assets/66555404/65a816b1-e2ab-4058-87a8-b33cc2afbb43)

V7.2
![B](https://github.com/avrdudes/avrdude/assets/66555404/1cd39d13-e5f8-4ca2-839f-dd00433cfde1)

V7.1
![v71](https://github.com/avrdudes/avrdude/assets/66555404/e87fdfab-93a6-4bce-b5d9-bd55a1f8aed0)

V6.3
![v63](https://github.com/avrdudes/avrdude/assets/66555404/47f266ba-9e50-4b0b-a40d-d91103664017)

**Comment by mcuee** [comment]
If you want to contribute to avrdude project and make it faster for your prefered programmer, then you may want to spend some efforts to see where avrdude 7.3 gets slower.

As of now, there is not much moving in the AVR910 front, probably even since 2011 (https://github.com/avrdudes/avrdude/commit/d742827da15a8e6165d956735ec48f1acff575e3). So yes PR is welcome to improve AVR910 support.
https://github.com/avrdudes/avrdude/commits/main/src/avr910.c

**Reply by flyandancexo** [reply]
If you try to do a comparison upload test between 7.3 and 7.2 using your own programmer, you should see a significant upload speed difference between these 2 versions. 7.3 is back 6.3's speed level. Is it a traditional to slow things down on x.3 version?

**Reply by mcuee** [reply]
I did not notice the speed drops myself. But I will try out some programmers (not AVR910) to see if there are regressions in terms of speed.
* https://github.com/avrdudes/avrdude/discussions/1022

**Reply by flyandancexo** [reply]
forget about it. I figured it out. I downloaded the MW version, and it's the cause of the the slow down.

**Reply by mcuee** [reply]
Interesting. I will check MSVC64 vs mingw64 speed as well for the programmers I have. Thanks for reporting back.

**Reply by mcuee** [reply]
I got similar results using usbasp, msvc64 build is faster than mingw64 build.
* https://github.com/avrdudes/avrdude/discussions/1022#discussioncomment-8537554

**Comment by flyandancexo** [comment]
> I did not notice the speed drops myself. But I will try out some programmers (not AVR910) to see if there are regressions in terms of speed.
> 
> * [Flash Read and Write Speed Benchmark #1022](https://github.com/avrdudes/avrdude/discussions/1022)

I have doubled checked. 73 is slow.
![Untitled-1](https://github.com/avrdudes/avrdude/assets/66555404/c49f6bb7-b9b5-45a7-8fdf-ee7e50d2262d)

Also I notice 73 can't use AVR910, but 72 is fine. 

```
C:\AVR\avrdude\v72\avrdude.exe -c AVR910 -p m128 -b 1000000 -x devcode=0x11 -P COM3 -U flash:w:"RD128K.txt":r -v
C:\AVR\avrdude\v73\avrdude.exe -c avr910 -p m128 -b 1000000 -x devcode=0x11 -P COM3 -U flash:w:"RD128K.txt":r -v
```

**Comment by stefanrueger** [comment]
> forget about it. I figured it out. I downloaded the MW version, and it's the cause of the the slow down.

Closing as it appears to be a false alarm
